src/core/reactor.cc

```diff
@@ -2218,6 +2219,41 @@ void reactor::kill(pid_t pid, int sig) {
     ret.throw_if_error();
 }
 
+future<std::optional<struct group>> reactor::getgrnam(std::string_view name, char *buf, size_t buflen) {
+    return futurize_invoke([name, buf, buflen, this] {
+        return _thread_pool->submit<syscall_result_extra<std::optional<struct group>>>([name = sstring(name), buf, buflen] {
+            errno = 0;
+            struct group grp;
+            struct group *result;
+            memset(&grp, 0, sizeof(struct group));
+            int ret = ::getgrnam_r(name.c_str(), &grp, buf, buflen, &result);
+            if (!result) {
+                return wrap_syscall(ret, std::optional<struct group>(std::nullopt));
+            }
+            return wrap_syscall(ret, std::optional<struct group>(*result));
+        }).then([] (syscall_result_extra<std::optional<struct group>> sr) {
+            if (sr.result != 0) {
+                // getgrnam_r, in case of error, an error number is returned (here it is stored in result), 
+                // which is the same as the error code in errno
```

Comment:
probably this comment is not necessary. as this follows the common pattern of other wrappers.